contracts/ConfidentialERC20/ConfidentialERC20.sol

```diff
@@ -260,14 +276,18 @@ abstract contract ConfidentialERC20 is Ownable, IConfidentialERC20, IERC20Metada
         burnRqs[requestID] = BurnRq(account, amount);
     }
 
-    function _burnCallback(uint256 requestID, bool decryptedInput) public virtual onlyGateway {
+    error InsufficientBalanceToBurn(address account, uint64 burnAmount);
+
+    function _burnCallback(uint256 requestID, bool hasEnoughBalance) public virtual onlyGateway {
         BurnRq memory burnRequest = burnRqs[requestID];
         address account = burnRequest.account;
         uint64 amount = burnRequest.amount;
-        if (!decryptedInput) {
-            revert("Decryption failed");
```

Comment:
This was a very misleading error message

Reply:
Hey how does the if statement work we get ebool from the hadSufficient function